src/v/resource_mgmt/storage.cc

```diff
@@ -92,13 +92,28 @@ ss::future<> disk_space_manager::stop() {
         _storage_node->local().unregister_disk_notification(
           node::disk_type::data, _data_disk_nid);
     }
+    _as.request_abort();
     _control_sem.broken();
     co_await _gate.close();
 }
 
 ss::future<> disk_space_manager::run_loop() {
     vassert(ss::this_shard_id() == run_loop_core, "Run on wrong core");
 
+    // Always wait for the trim interval after startup to ensure that
+    // partitions are fully started. Known sources of skew early in the node
+    // startup phase:
+    // 1. controller_backend::start() doesn't wait for the first reconciliation
+    // round, so not all partitions may yet exist in the partition manager.
+    // 2. partition::start() doesn't wait for STMs to start, so
```

Comment:
I think point 2. is not entirely correct here. The `partition::start()` waits for the state machines to start but it doesn't wait for the state machines to be fully replied.

Reply:
Yes you are right, it is more complicated than I thought previously. `partition::start()` waits for local snapshots to be applied, but for some STMs (in particular, rm_stm and archival_meta_stm) additional things must happen before they start returning sensible max_collectible_offset.